packages/vite-node/src/client.ts

```diff
@@ -269,6 +277,10 @@ export class ViteNodeRunner {
 
     const request = async (dep: string) => {
       const [id, depFsPath] = await this.resolveUrl(`${dep}`, fsPath)
+      const depMod = this.moduleCache.getByModuleId(depFsPath)
+      depMod.importers.add(moduleId)
```

Comment:
I think this is already done on line 200?

Reply:
This is not always the case. The `dependencyRequest` is an async function that will not be executed immediately.

Reply:
Why does it matter? There is nothing async between this calls

Reply:
The current test case will be stuck if I remove this line

Reply:
<img width="977" alt="image" src="https://github.com/vitest-dev/vitest/assets/29533304/c9516d26-23d0-44a6-837f-dc3b8dd149b8">

The execution sequence looks like this

Reply:
Interesting 🤔 Ok, then, should be fine